### PR TITLE
Add ability to delete and re-draft a comment with no replies + markdown infos + unicode emojis

### PR DESCRIPTION
**Pull request by kimsible**
- Hide mention of deleted comments without replies ;
- New delete and re-draft action on a comment **only on comments without replies** ;
- Move delete, delete and re-draft actions on options menu ;
- Add icons on delete, delete and re-draft and report actions in options menu ;
- Rename `Reply` value of add comment button to `Comment` **only for comment main thread textarea** ;
- Add Markdown help icon :
- Add uni-code emojis native display (depending on Operating System) https://apps.timwhitlock.info/emoji/tables/unicode.

Comments-markdown

### CODE REVIEW

File: client/src/app/+videos/+video-watch/comment/video-comment.component.html [hunk 1]
```diff
@@ -58,7 +57,7 @@
              class="comment-date">{{ comment.createdAt | myFromNow }}</a>
         </div>
 
-        <div *ngIf="comment.isDeleted" class="comment-html comment-html-deleted">
```
Comment (kimsible) [inline client/src/app/+videos/+video-watch/comment/video-comment.component.html:61]: Deleted because duplication of conditional with the parent container

**Comment by rigelk** [comment]
Nice! I like the markdown icon but at the same time I'm not too sure about it's position. Usually this is right aligned in the input, and greyed like the input border.

**Comment by kimsible** [comment]
> Nice! I like the markdown icon but at the same time I'm not too sure about it's position. Usually this is right aligned in the input, and greyed like the input border.

Yes, but we have the extend native icon of textarea on the right. 
We also can do as GiThub does, but we loose the one line textarea and the UI becomes more complex : 
image

**Comment by kimsible** [comment]
@rigelk 
markdown-on-right

**Comment by kimsible** [comment]
The last commit adds unicode emojis to markdown : 
emoji

**Comment by rigelk** [comment]
@kimsible that is what I though, yes :)

Could you please fix https://github.com/Chocobozzz/PeerTube/pull/3046/checks?check_run_id=961331548#step:6:126?

**Comment by kimsible** [comment]
> @kimsible that is what I though, yes :)
> 
> Could you please fix https://github.com/Chocobozzz/PeerTube/pull/3046/checks?check_run_id=961331548#step:6:126?

Thanks, I'm working on it.

For the emojis, I don't know if we must include an emoji collection into PeerTube, in the best way, yes, because some people do not have every one on their operating system (especially on desktop linux and windows). 

What do we do ?

**Comment by rigelk** [comment]
> For the emojis, I don't know if we must include an emoji collection into PeerTube, in the best way, yes, because some people do not have every one on their operating system (especially on desktop linux and windows).
> 
> What do we do ?

I'm inclined to look at what Mastodon does, where they adopted a full emoji management interface that is highly extensible. Until then, only including a light set like `markdown-it-emoji` provides sounds like a possible option (+25kB for the full version or +7kB for the light version sounds *reasonable*).

It's not like PeerTube is only focusing on text and thus very reliant on emojis, but emojis have their place too in the comment palette (i.e. [an issue has already considered video reactions](https://github.com/Chocobozzz/PeerTube/issues/2212) - a much heavier development - which proves there is interest for richer expression forms to react to videos).

**Comment by kimsible** [comment]
> only including a light set like markdown-it-emoji provides sounds like a possible option (+25kB for the full version or +7kB for the light version sounds reasonable).

Yes, the light version sounds reasonable thanks.

> It's not like PeerTube is only focusing on text and thus very reliant on emojis, but emojis have their place too in the comment palette (i.e. an issue has already considered video reactions - a much heavier development - which proves there is interest for richer expression forms to react to videos).

I think you meant this PR https://github.com/Chocobozzz/PeerTube/issues/3026 ?

**Comment by rigelk** [comment]
@kimsible both of them actually, though they are just issues, not PRs.

**Comment by kimsible** [comment]
> @kimsible both of them actually, though they are just issues, not PRs.

Ok @rigelk , so you mean adding a Popover with emojis ? 

Otherwise, I'm not sure (since I'm not an expert of Typescript) but light version emoji of markdown-it seems to not be implemented to `@types/markdown-it-emoji` https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/markdown-it-emoji/index.d.ts, the commonjs way is `require('markdown-it-emoji/light')`, do you know how to do with Typescript ?

**Comment by rigelk** [comment]
> > @kimsible both of them actually, though they are just issues, not PRs.
> 
> Ok @rigelk , so you mean adding a Popover with emojis ?

no, I didn't mean anything like this. The issue I quoted was merely an example of how text is arguably not the only response type we should consider, as an argument in favor of including emojis (since you asked if we *should* include emojis).

Back on your current error, leave it at the typical `require` for the moment. We could always write types, but for now it's more work than it's worth.

**Comment by kimsible** [comment]
The last commits uses Commonjs require to include markdown-it emoji plugin and to get light emoji list.
It also refactor the Markdown Pophover : 
image

image

File: client/package.json [hunk 2]
```diff
@@ -61,6 +61,7 @@
     "@types/linkifyjs": "^2.1.2",
     "@types/lodash-es": "^4.17.0",
     "@types/markdown-it": "^10.0.1",
+    "@types/markdown-it-emoji": "^1.4.0",
```
Comment (kimsible) [inline client/package.json:64]: TODO : to remove

File: client/yarn.lock [hunk 3]
```diff
@@ -1571,7 +1571,14 @@
   dependencies:
     "@types/node" "*"
 
-"@types/markdown-it@^10.0.1":
+"@types/markdown-it-emoji@^1.4.0":
```
Comment (kimsible) [inline client/yarn.lock:1574]: TODO : to remove

**Comment by kimsible** [comment]
Thanks for the review @Chocobozzz, I've fixed / refactored all your comments.

**Comment by kimsible** [comment]
@Chocobozzz maybe this PR should have been merged before changing localization system, we need to re-pass on all files

**Comment by rigelk** [comment]
@kimsible templates are not touched, just their associated components logic as the I18n service model is replaced.

**Comment by kimsible** [comment]
> @kimsible templates are not touched, just their associated components logic as the I18n service model is replaced.
Thanks @rigelk, ok, in this PR we have some `i18n-title` and `i18n` tags, so we leave them as they are ?

**Comment by rigelk** [comment]
Yes, since templates are not affected.

**Comment by Chocobozzz** [comment]
I'll merge this PR myself.

**Comment by kimsible** [comment]
> I'll merge and fix conflicts of this PR myself.

Thanks @Chocobozzz, I've already merged it but maybe you would like to check if I missed something.

**Pull request merged**